src/components/DepositWidget/useRowActions.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useRef, SetStateAction, Dispatch } from 'react'
+import { toast } from 'react-toastify'
+import BN from 'bn.js'
+
+import { log } from 'utils'
+import { depositApi, erc20Api } from 'api'
+import { Mutation, TokenBalanceDetails, TxOptionalParams, Receipt } from 'types'
+import { HIGHLIGHT_TIME, ALLOWANCE_MAX_VALUE, ZERO } from 'const'
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import { TxNotification } from 'components/TxNotification'
+import { formatAmount, formatAmountFull } from 'utils'
+
+interface Params {
+  balances: TokenBalanceDetails[]
+  setBalances: Dispatch<SetStateAction<TokenBalanceDetails[]>>
+}
+
+interface Result {
+  enableToken: (tokenAddress: string) => Promise<void>
+  deposit: (amount: BN, tokenAddress: string) => Promise<void>
+  requestWithdraw: (amount: BN, tokenAddress: string) => Promise<void>
+  claim: (tokenAddress: string) => Promise<void>
+}
+
+const txOptionalParams: TxOptionalParams = {
+  onSentTransaction: (receipt: Receipt): void => {
+    if (receipt.transactionHash) {
+      toast.info(<TxNotification txHash={receipt.transactionHash} />)
+    } else {
+      console.error(`Failed to get notification for tx ${receipt.transactionHash}`)
+    }
+  },
+}
+
+export const useRowActions = (params: Params): Result => {
+  const { balances, setBalances } = params
+  const { userAddress } = useWalletConnection()
+  const contractAddress = depositApi.getContractAddress()
+  const mounted = useRef(true)
+  useEffect(() => {
+    return function cleanUp(): void {
+      mounted.current = false
+    }
+  }, [])
+
+  function _getToken(tokenAddress: string): TokenBalanceDetails {
+    return balances.find(({ address }) => address === tokenAddress)
+  }
+
+  function _updateToken(tokenAddress: string, updateBalances: Mutation<TokenBalanceDetails>): void {
+    setBalances(balances =>
```

Comment:
You can move `if mounted`check here, then there'll be no need to include it in any other places in this file.

Or, and this is crazy, we can agree on #132 and implement one of the proposals.

Reply:
Addressed in 5de6de3070cb18b1c4baf5b823c0f2c50e8f06d3

Yes, let's discuss the refactor in that issue.